src/nickel_error.rs

```diff
@@ -25,8 +25,8 @@ impl<D: Send + 'static + Sync> NickelError<D> {
     /// use nickel::status::StatusCode;
     ///
     /// # #[allow(dead_code)]
-    /// fn handler<D>(_: &mut Request<D>, res: Response<D>) -> MiddlewareResult<D> {
-    ///     Err(NickelError::new(res, "Error Parsing JSON", StatusCode::BadRequest))
+    /// fn handler<D: Send + 'static + Sync>(_: &mut Request<D>, res: Response<D>) -> MiddlewareResult<D> {
```

Comment:
`D` was always `Send + Sync` IIRC, but now it needs to be `'static` too?  If that's a consequence of async, so be it.  (It would explain why the lifetimes got so much simpler.)

Reply:
Yeah, async scrambles lifetimes. Overall type signatures are simpler though.